tests/integration/test_charm.py

```diff
@@ -25,23 +25,22 @@
 AUTH_PROXY_REQUIRER = "auth-proxy-requirer"
 
 
-async def get_unit_address(ops_test: OpsTest, app_name: str, unit_num: int) -> str:
-    """Get private address of a unit."""
-    status = await ops_test.model.get_status()  # noqa: F821
-    return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
-
-
-async def get_app_address(ops_test: OpsTest, app_name: str) -> str:
-    """Get address of an app."""
-    status = await ops_test.model.get_status()  # noqa: F821
-    return status["applications"][app_name]["public-address"]
+async def get_k8s_service_address(ops_test: OpsTest, service_name: str, lightkube_client: Client) -> Optional[str]:
+    """Get the address of a LoadBalancer Kubernetes service using kubectl."""
+    try:
+        result = lightkube_client.get(Service, name=service_name, namespace=ops_test.model.name)
+        ip_address = result.status.loadBalancer.ingress[0].ip
```

Comment:
won't this fail if ingress is empty?

Reply:
The tests are expected to fail if there is no ingress